modules/plugin/chowdsp_plugin_utils/SharedUtils/chowdsp_GlobalPluginSettings.h

```diff
@@ -79,6 +79,8 @@ class GlobalPluginSettings
 
     static constexpr SettingID settingsTag = "plugin_settings";
 
+    inline static juce::CriticalSection lock;
```

Comment:
I think the `CriticalSection` objects are typically used as non-static member variables. From the JUCE docs:
> In almost all cases you'll want to declare your [CriticalSection](https://docs.juce.com/master/classCriticalSection.html) as a member variable. Occasionally you may want to declare one as a static variable, but in that case the usual C++ static object order-of-construction warnings should be heeded.

(sorry I had this comment in my original review, but it got deleted somehow)

Reply:
I saw this as well, I'll look into it again, but when understanding was that the `CriticalSection` had to be a static variable as one instance of the GlobalPluginSettings needed to access that same Critical Section to lock it from all other instances. Let me know if that makes sense to you as well.

Initially I was using the CriticalSection as a non-static variable, what I saw was that both threads were entering the `initialise` function at the same time and overwriting the file. They were both locking their own respective critical section non-static member variables but had no way of communicating between the two threads when reads and writes were happening. It's possible there is a better way to solve this without using a static CriticalSection as well, I just couldn't think of one.